plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/File.tsx

```diff
@@ -0,0 +1,103 @@
+import { CatalogItem } from "@web/extensions/sidebar/core/processCatalog";
+import { Button, Icon } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useMemo } from "react";
+
+export type Props = {
+  item: CatalogItem;
+  addedDatasetIds?: string[];
+  nestLevel: number;
+  selected: boolean;
+  onDatasetAdd: (dataset: CatalogItem) => void;
+  onOpenDetails?: (item?: CatalogItem) => void;
+  onSelect?: (id: string) => void;
+};
+
+const File: React.FC<Props> = ({
+  item,
+  addedDatasetIds,
+  nestLevel,
+  selected,
+  onDatasetAdd,
+  onOpenDetails,
+  onSelect,
+}) => {
+  const handleClick = useCallback(() => {
+    onDatasetAdd(item);
+  }, [item, onDatasetAdd]);
+
+  const handleOpenDetails = useCallback(() => {
+    if (item.type === "group") return;
+    onOpenDetails?.(item);
+    onSelect?.(item.id);
+  }, [item, onOpenDetails, onSelect]);
+
+  const addDisabled = useMemo(() => {
+    return !!addedDatasetIds?.find(id => item?.type === "item" && id === item.id);
+  }, [addedDatasetIds, item]);
+
+  return (
+    <>
+      {item.type === "item" && (
+        <Wrapper nestLevel={nestLevel} selected={selected}>
+          <NameWrapper onClick={handleOpenDetails}>
+            <Icon icon="file" size={20} />
+            <Name>{item.cityName ?? item.name}</Name>
+          </NameWrapper>
+          <Button
+            type="link"
+            icon={<StyledIcon icon="plusCircle" selected={selected ?? false} />}
+            onClick={handleClick}
+            disabled={addDisabled}
+          />
+        </Wrapper>
+      )}
+    </>
+  );
```

Comment:
Remember, you can write this like
```suggestion
  return item.type === "item" && (
        <Wrapper nestLevel={nestLevel} selected={selected}>
          <NameWrapper onClick={handleOpenDetails}>
            <Icon icon="file" size={20} />
            <Name>{item.cityName ?? item.name}</Name>
          </NameWrapper>
          <Button
            type="link"
            icon={<StyledIcon icon="plusCircle" selected={selected ?? false} />}
            onClick={handleClick}
            disabled={addDisabled}
          />
        </Wrapper>
      );
```

Reply:
Don't need the fragments and it simplifies/cleans up the code a bit

Reply:
I don't think that this is possible. ReactElement should be returned.
The following error will occure:
`Type 'boolean' is not assignable to type 'ReactElement<any, any>'.`

Reply:
Done!